### PR TITLE
Conflict resolution of upstream merge 2.5.1

### PR DESCRIPTION
**Pull request by cloudw**
Resolving conflicts from upstream.

Most of conflicts are result of netflix/metaflow embracing black formatting 🎉 

Major changes that resulted in code changes:
- https://github.com/Netflix/metaflow/pull/929 Related imports in KFP plug-in are rewrote
- https://github.com/Netflix/metaflow/pull/580 All datastore usage in KFP plug-in are rewrote. This touches all decorator, kfp cli and kfp foreach splits.
- Coverage functionalities are removed as it is removed from upstream as well. ⚠️ May result in test coverage drop for customers using flow in unit tests.
- Other conflict resolutions

### CODE REVIEW

**Review by alexlatchford — APPROVED** [review]

**Comment by alexlatchford** [comment]
I'd say we get the tests working before merging this, looks like there are still some `<<<` lines in there based on the test details :)

File: metaflow/current.py [hunk 1]
```diff
@@ -84,10 +62,6 @@ def get(self, key, default=None):
     def is_running_flow(self):
         return self._is_running
 
-    @property
-    def flow(self) -> FlowSpec:
-        return self._flow
```
Comment (talebzeghmi) [inline metaflow/current.py:89]: this is missing

Comment (cloudw) [reply metaflow/current.py:89]: I can add this back for sure. Meanwhile I don't see a code usage for this function now.

Comment (cloudw) [reply metaflow/current.py:89]: Added back!

**Comment by cloudw** [comment]
Closing in favor of merge_upstream_2_5_3 branch

**Pull request closed without merging**